sklearn/mixture/dirichlet_gaussian_mixture.py

```diff
@@ -0,0 +1,338 @@
+"""Dirichlet Process Bayesian Gaussian Mixture Model."""
+# Author: Wei Xue <xuewei4d@gmail.com>
+#         Thierry Guillemot <thierry.guillemot.work@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.special import digamma, betaln
+
+from ..utils.validation import check_is_fitted
+from .gaussian_mixture import _compute_log_det_cholesky
+from .bayesian_mixture import BayesianGaussianMixture, _log_wishart_norm
+
+
+class DirichletGaussianMixture(BayesianGaussianMixture):
+    """Dirichlet Process Bayesian Gaussian Mixture.
+
+    An infinite mixture model with the Dirichlet Process as a prior
+    distribution on the number of clusters. In practice the
+    approximate inference algorithm uses a truncated distribution with
+    a fixed maximum number of components, but almost always the number
+    of components actually used depends on the data.
+
+    Stick-breaking Representation of a Gaussian mixture model
```

Comment:
Stick-breaking Representation of a Gaussian mixture model
 +    probability distribution

this is not a sentence